pdfiumandroid/src/main/java/io/legere/pdfiumandroid/util/PdfiumNativeSourceBridge.kt

```diff
@@ -0,0 +1,28 @@
+package io.legere.pdfiumandroid.util
+
+import io.legere.pdfiumandroid.BuildConfig
+import io.legere.pdfiumandroid.PdfiumSource
+
+internal class PdfiumNativeSourceBridge(private val source: PdfiumSource) {
+
+    private var buffer: ByteArray? = null
+
+    fun read(position: Long, size: Long): Int =
+        try {
+            require(size <= Int.MAX_VALUE) { "size is too large" }
```

Comment:
This is very unlikely to happen and I think long type was chosen for array size in Pdfium by mistake. But if it does happen, there's nothing we can do, there's no way to allocate byte buffer with size > Int.MAX_VALUE. But just in case, to avoid native crashes in runtime, we check before we cast